src/metabase/query_processor/util/add_alias_info.clj

```diff
@@ -186,7 +186,7 @@
 
 (defn- exports [query]
   (into #{} (mbql.u/match (dissoc query :source-query :source-metadata :joins)
-              [(_ :guard #{:field :expression :aggregation}) _ (_ :guard (every-pred map? ::position))])))
+              [(_ :guard #{:field :expression :aggregation-options}) _ (_ :guard (every-pred map? ::position))])))
```

Comment:
So this wasn't a typo. `[:aggregation <index>]` is a reference to the aggregation at that index. Example

```clj
;; SELECT COUNT(*) AS "count" FROM "my_table" ORDER BY "count" DESC 
{:source-table 2
 :aggregation [[:count]]
 :order-by [[:desc [:aggregation 0]]]}
```

I think the idea was that `:aggregation` was supposed to come back with the appropriate info including the `::position` and stuff like that. That being said this clearly wasn't doing the right thing and wasn't working as intended, so I'm cool with doing something different.